app/validation/validator.py

```diff
@@ -74,6 +77,27 @@ def _validate_json_against_schema(self, json_to_validate):
         except SchemaError as e:
             return '{}'.format(e)
 
+    def _validate_schema_contain_required_metadata(self, schema):
+        # Ensure that metadata used within the schema are explicitly defined.
+
+        errors = []
+
+        default_metadata = ['user_id', 'period_id']  # Metadata that must be present in all schemas
+        required_metadata = schema['required_metadata']  # Metadata required within the schema.
+
+        # Find all words that precede with "metadata['"
+        parsed_metadata = re.findall(r"(?<=metadata\[\')\w+", str(schema))
+
+        for metadata in set(parsed_metadata):
+            if metadata not in required_metadata:
+                errors.append(self._error_message('Metadata - {} not specified in required_metadata field'.format(metadata)))
+
+        for metadata in required_metadata:
```

Comment:
I don't think this check adds any value and could cause author issues in the future.

If somebody were to add some required metadata I'm not sure it makes sense to force them to make use of it.

Reply:
I get your point but I thought the point was so that all metadata needed to launch and submit the survey must be specified but will we get any value from allowing metadata to be specified that isn't needed by runner?. 

If it is metadata that is used by runner but not needed by the `schema_context`, should it even be specified here in the first place?

Reply:
I think this check is valid. The point of this PR is to fail early in runner. If we define metadata we don't then use there could be situations where we fail needlessly in runner.